src/main/java/seedu/address/logic/parser/FindCommandParser.java

```diff
@@ -156,4 +113,82 @@ private double findUpperBound(String salaryArgs) {
         return Double.parseDouble(salaryArgs.split(" - ", 2)[1]);
     }
 
+    private void parseIdKeyword(ArrayList<Predicate<Person>> predicateList,
```

Comment:
The parse commands seems to have lots of repeated code, in particular this few lines. Perhaps you can create a helper function like `checkTrimmedArgs` to reduce repetition?
```
        ParserUtil.checkFindArgs(trimmedArgs);
        if (trimmedArgs.isEmpty()) {
            throw new ParseException(
                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
        }
```

Reply:
Ok! I did this and also moved some methods into the ParserUtil class